commit 01d9157013729859fdb7470887d78d67a3cdf6b9
Author: Christopher Faulet <cfaulet@haproxy.com>
Date:   Fri Apr 19 15:26:01 2019 +0200

    BUG/MINOR: da: Get the request channel to call CHECK_HTTP_MESSAGE_FIRST()
    
    Since the commit 89dc49935 ("BUG/MAJOR: http_fetch: Get the channel depending on
    the keyword used"), the right channel must be passed as argument when the macro
    CHECK_HTTP_MESSAGE_FIRST is called.
    
    This patch must be backported to 1.9.
    
    (cherry picked from commit f48552f2c10a2f956d7bd1eb02a6d694d2b5c5d3)
    Signed-off-by: Christopher Faulet <cfaulet@haproxy.com>
    (cherry picked from commit 2a38fa09b588d8b30fabc77282e66ef613336ee7)
    Signed-off-by: Christopher Faulet <cfaulet@haproxy.com>

diff --git a/src/da.c b/src/da.c
index 685a79d1..cd4050bf 100644
--- a/src/da.c
+++ b/src/da.c
@@ -293,7 +293,7 @@ static int da_haproxy_fetch(const struct arg *args, struct sample *smp, const ch
 		return 1;
 	}
 
-	CHECK_HTTP_MESSAGE_FIRST();
+	CHECK_HTTP_MESSAGE_FIRST((smp->strm ? &smp->strm->req : NULL));
 	smp->data.type = SMP_T_STR;
 
 	/**
